pkg/operator/ceph/cluster/cluster.go

```diff
@@ -482,3 +350,69 @@ func (c *cluster) notifyChildControllerOfUpgrade() error {
 
 	return nil
 }
+
+// Validate the cluster Specs
+func (c *ClusterController) preClusterStartValidation(cluster *cluster, clusterObj *cephv1.CephCluster) error {
```

Comment:
Looks better yes, thanks.

Reply:
Actually, this feels a bit odd to have it part of `cluster` since at this point the cluster is not even initialized. Also, this would mean changing all the parent calls to be part of `c *cluster`.